packages/dd-trace/src/appsec/iast/telemetry/index.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const telemetryMetrics = require('../../../telemetry/metrics')
+const telemetryLogs = require('./log')
+const { Verbosity, isDebugAllowed, isInfoAllowed, getVerbosity } = require('./verbosity')
+const { initRequestNamespace, finalizeRequestNamespace, globalNamespace } = require('./namespaces')
+
+class Telemetry {
+  configure (config, verbosity) {
+    // in order to telemetry be enabled, tracer telemetry and metrics collection have to be enabled
+    this.enabled = config && config.telemetry && config.telemetry.enabled && config.telemetry.metrics
+    this.verbosity = this.enabled ? getVerbosity(verbosity) : Verbosity.OFF
+
+    if (this.enabled) {
+      telemetryMetrics.manager.set('iast', globalNamespace)
+    }
+
+    telemetryLogs.start()
+  }
+
+  stop () {
+    this.enabled = false
+    telemetryMetrics.manager.delete('iast')
+
+    telemetryLogs.stop()
+  }
+
+  isEnabled () {
+    return this.enabled
+  }
+
+  isDebugEnabled () {
+    return this.isEnabled() && isDebugAllowed(this.verbosity)
+  }
+
+  isInformationEnabled () {
+    return this.isEnabled() && isInfoAllowed(this.verbosity)
+  }
+
+  onRequestStart (context) {
+    if (this.isEnabled() && this.verbosity !== Verbosity.OFF) {
+      initRequestNamespace(context)
+    }
+  }
+
+  onRequestEnd (context, rootSpan) {
+    if (this.isEnabled() && this.verbosity !== Verbosity.OFF) {
+      finalizeRequestNamespace(context, rootSpan, this.namespace)
```

Comment:
`this.namespace` is never defined. Besides that, `finalizeRequestNamespace ` does not expect a third argument since it is getting it from context. [finalizeRequestNamespace](https://github.com/DataDog/dd-trace-js/pull/2805/files#diff-ebc7da71cb34d449a137d39d0634ce503c29b3099f278a99790f27b6cd767390R22)